dali/pipeline/graph_descr.cc

```diff
@@ -71,358 +73,444 @@ void CheckOpConstraints(const OpSpec &spec) {
       + " outputs, but was passed " + std::to_string(spec.NumOutput()/num_input_sets) + ".");
 }
 
+
+DALIOpType DeviceStringToOpType(std::string device) {
+  if (device == "gpu") {
+    return DALIOpType::DALI_GPU;
+  } else if (device == "cpu") {
+    return DALIOpType::DALI_CPU;
+  } else if (device == "mixed") {
+    return DALIOpType::DALI_MIXED;
+  } else if (device == "support") {
+    return DALIOpType::DALI_SUPPORT;
+  }
+  DALI_FAIL("Unsupported device type: " + device + ".");
+}
+
+DALITensorDevice DeviceStringToTensorStorage(std::string inout_device) {
```

Comment:
1. `inout_device`? Is it misspelled `input_device` or, more familar `io_device`?
2. string passed by value, but not moved anywhere inside
3. Inconsistent name -> should be `DeviceStringToTensorDevice` or, better, `ParseTensorDevice` (or sth. else if you decide to rename DALITensorDevice).

Reply:
I meant input and output, because it is for both inputs and outputs of pipeline.

Done.